plugins/git/gitblame.nix

```diff
@@ -0,0 +1,99 @@
+{
+  pkgs,
+  config,
+  lib,
+  ...
+}:
+with lib; let
+  cfg = config.plugins.gitblame;
+  helpers = import ../helpers.nix {inherit lib;};
+in {
+  options = {
+    plugins.gitblame = {
+      enable = mkEnableOption "gitblame";
+
+      package = helpers.mkPackageOption "gitblame" pkgs.vimPlugins.git-blame-nvim;
+
+      messageTemplate = mkOption {
```

Comment:
Thanks for the pr! Just a note though, don't have these as the default! Use the helper functions in helpers.nix in order to define options.

Defining them like this leads to quickly filling up the generated config, since we are redefining every default option.

Reply:
Sure, I'll update those. I used mkOption because I was looking at how other plugins are written 😅

Reply:
done.

though for one option, `display_virtual_text`, I had to keep `mkOption`.
The plugin expects 1, 0 or null with default being 1 (if null).

when using `helpers.defaultNullOpts.mkNullable (types.nullOr types.bool) true`, I am not able to convert it to 1 or 0 in the config

Reply:
You need to do it manually with something like 
```
display_virtual_text = let disp = cfg.displayVirtualText in 
   helpers.ifNonNull' cfg.displayVirtualText (if disp then 1 else 0);
```

Reply:
That seems to have worked :+1:

Reply:
resolving this as all the options are now using helpers